### PR TITLE
Phase1 issue v2

### PR DESCRIPTION
**Pull request by binh-vu**

### CODE REVIEW

File: www/src/models/mineralSite/MineralSite.ts [hunk 1]
```diff
@@ -16,10 +16,10 @@ export type FieldEdit =
...
   | { field: "depositType"; observedName: string; normalizedURI: string }
   | {
-      field: "grade";
-      value: number;
-      commodity: string;
-    }
+    field: "grade";
+    value: number;
+    commodity: string;
+  }
```
Comment (binh-vu) [inline www/src/models/mineralSite/MineralSite.ts:22]: Please setup auto formatting on vscode. I already mentioned this multiple times

File: www/src/models/mineralSite/MineralSite.ts [hunk 2]
```diff
@@ -155,37 +155,69 @@ export class DraftCreateMineralSite extends MineralSite {
...
+      switch (edit.field) {
+        case "name":
+          draftSite.name = edit.value;
+          break;
+        case "location":
+          draftSite.locationInfo.location = edit.value;
+          break;
+        case "depositType":
+          draftSite.depositTypeCandidate = [
+            new CandidateEntity({
```
Comment (binh-vu) [inline www/src/models/mineralSite/MineralSite.ts:190]: I may not be cleared but we don't want to add an edit here as it duplicates the code.

File: www/src/models/mineralSite/MineralSite.ts [hunk 3]
```diff
@@ -155,37 +155,69 @@ export class DraftCreateMineralSite extends MineralSite {
...
+          break;
+        default:
+          throw new Error(`Unhandled edit field: ${edit}`);
+      }
+    }
+
+    return draftSite;
   }
 
+
```
Comment (binh-vu) [inline www/src/models/mineralSite/MineralSite.ts:220]: again formatting issue

File: www/src/pages/home/editDedupSite/EditDedupMineralSite.tsx [hunk 4]
```diff
@@ -39,7 +40,18 @@ export const EditDedupMineralSite = observer(({ dedupSite, commodity }: EditDedu
       message.success("Ungrouping was successful!");
     }
   };
+  const getUserColor = (username: string) => {
+    let hash = 0;
+    for (let i = 0; i < username.length; i++) {
+      hash = username.charCodeAt(i) + ((hash << 5) - hash);
+      hash = hash & hash;
+    }
+    const hue = Math.abs(hash % 360);
```
Comment (binh-vu) [inline www/src/pages/home/editDedupSite/EditDedupMineralSite.tsx:49]: format issue again

File: www/src/models/mineralSite/MineralSite.ts (hunk 1 above)
Comment (NithyashreeGS30) [reply www/src/models/mineralSite/MineralSite.ts:22]: I have already done that

File: www/src/pages/home/editDedupSite/EditDedupMineralSite.tsx (hunk 4 above)
Comment (binh-vu) [reply www/src/pages/home/editDedupSite/EditDedupMineralSite.tsx:49]: move this function out

**Pull request closed without merging**